hazelcast/config.py

```diff
@@ -216,21 +216,15 @@ def unique_key(self):
 
     @unique_key.setter
     def unique_key(self, value):
-        if get_attr_name(QueryConstants, value):
-            self._unique_key = value
-        else:
-            raise TypeError("unique_key must be of type QueryConstants")
-    
+        try_to_set_enum_value(self, value, QueryConstants, "_unique_key")
```

Comment:
nit: I'd prefer `try_to_set_enum_value` to have no side effects and just convert the `int|string` into the expected `string`. This way there will be no need to pass the stringly typed field name (i.e. `"_unique_key"`) into the function. As another side effect, the function can be tested in an easier way. It could look something like the following:
```python
self._unique_key = try_to_get_enum_value(self, value, QueryConstants)
```

Reply:
Thanks for the suggestion. Changed to something similar with no side effects

Reply:
Thanks for doing the change. The only improvement I can think of is renaming `try_to_set_enum_value` into `try_to_get_enum_value` or something similar. That's because the function no longer sets the value - it only converts it.

Reply:
Oh, I missed that. Updated it

Reply:
Thanks